### PR TITLE
Increase Core folder test coverage

### PR DESCRIPTION
**Pull request by byorda-glb**
### Purpose

Increase Core folder test coverage including a few tests in the AuthenticationManager class.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

Aaron Tang
Michael Kirschner

### FYIs

Alfredo Pozo

### CODE REVIEW

**Comment by byorda-glb** [comment]
image

**Comment by QilongTang** [comment]
LGTM

File: test/DynamoCoreTests/AuthenticationManagerTests.cs [hunk 1]
```diff
@@ -47,5 +57,56 @@ public void Login_CausesLoginMethodToBeInvokedOnAuthProvider()
...
+
+        [Test]
+        public void Property_Username()
+        {
+            var authProviderMock = new Mock<IAuthProvider>();
+            var authManager = new AuthenticationManager(authProviderMock.Object);
+
+            authProviderMock.Setup(x => x.Username).Returns("test");
+
+            Assert.AreEqual(authManager.Username, "test");
```
Comment (QilongTang) [inline test/DynamoCoreTests/AuthenticationManagerTests.cs]: Would you also assert the other public members here? e.g. `HasAuthProvider`

Comment (byorda-glb) [reply test/DynamoCoreTests/AuthenticationManagerTests.cs]: The other member appear as covered. Maybe by some other tests.

Comment (byorda-glb) [reply test/DynamoCoreTests/AuthenticationManagerTests.cs]: @QilongTang Should I add specific tests for those members, even though they are already covered by the current tests? (For example, HasAuthProvider is covered when testing the Username property)
image

Comment (QilongTang) [reply test/DynamoCoreTests/AuthenticationManagerTests.cs]: Sorry not meant for asking to create new tests. If you can cover asserting `HasAuthProvider` and `LoginState`, that's good enough

Comment (byorda-glb) [reply test/DynamoCoreTests/AuthenticationManagerTests.cs]: Two tests were added to check for the HasAuthProvider and LoginState properties.

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**